lib/rules/for-direction.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @fileoverview enforce "for" loop update clause moving the counter in the right direction.(for-direction)
+ * @author Aladdin-ADD<hh_2013@foxmail.com>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce \"for\" loop update clause moving the counter in the right direction.",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+
+        return {
+            ForStatement(node) {
+
+                /**
+                 * report an error.
+                 * @returns {void}
+                 */
+                function report() {
```

Comment:
Could this function be moved outside of the ForExpression visitor? As it is, this function is created every time a ForStatement is encountered in the code and it looks like a pure function (doesn't depend on the `node` from the ForStatement visitor function params).